velox/vector/ComplexVector.cpp

```diff
@@ -681,6 +681,125 @@ void RowVector::resize(vector_size_t newSize, bool setNotNull) {
   }
 }
 
+namespace {
+
+struct Wrapper {
```

Comment:
LOL, I can't think of a better name either. It's like a Peer/ParentDictionaryStates at the level of the traversal. Would it be appropriate to call it DictionaryVisitor lol.

Reply:
`DictionaryVisitor` is not appropriate IMO, a visitor is unique during the traversal, this is created per dictionary node.